src/EnergyPlus/UnitarySystem.cc

```diff
@@ -3825,7 +3825,7 @@ namespace UnitarySystems {
         // if part of ZoneHVAC:OutdoorAirUnit bypass most checks for connection to air loop or OASystem
         if (ZoneOAUnitNum > 0) OASysFound = true;
 
-        if (!AirLoopFound && !OASysFound) {
+        if (ZoneEquipment) {
```

Comment:
Use `ZoneEquipment` parameter.

Reply:
I wonder if there is an example file with UnitarySystem inside a ZoneHVAC:OutdoorAirUnit such that would test that this logic includes that configuration? Otherwise all these changes look correct to me.

Reply:
> I wonder if there is an example file with UnitarySystem inside a ZoneHVAC:OutdoorAirUnit such that would test that this logic includes that configuration? Otherwise all these changes look correct to me.

Both OutdoorAirUnit.idf and OutdoorAirUnitwithAirloopHVAC.idf have a CoilSystem:Cooling:DX in the ZoneHVAC:OutdoorAirUnit:EquipmentList, so I think this case is covered. I followed progress in the debugger using OutdoorAirUnit.idf, and it executed or skipped in the right places.

Reply:
OK, enough speculating whether these changes cover all cases. This branch is clean so is probably ready for testing the defect file and moving on.

Reply:
:+1: thanks for your commentary here @rraustad